src/emscripten-optimizer/istring.h

```diff
@@ -66,22 +68,24 @@ struct IString {
     typedef std::unordered_set<const char *, CStringHash, CStringEqual> StringSet;
     static StringSet* strings = new StringSet();
 
-    if (reuse) {
-      auto result = strings->insert(s); // if already present, does nothing
-      str = *(result.first);
-    } else {
-      auto existing = strings->find(s);
-      if (existing == strings->end()) {
+    auto existing = strings->find(s);
+
+    if (existing == strings->end()) {
+      // the StringSet is a global shared structure, which can be modified
+      // only on the main thread and only when other threads cannot race.
```

Comment:
"only on the main thread" comment is still correct? Are you just relying on tsan to catch violations of that rule?